### ISSUE TITLE
A null parameter should throw ArgumentNullException

### ISSUE THREAD

**Issue by tsahi** [SanderSade/Sander.KeyVaultCache#1]
at https://github.com/SanderSade/Sander.KeyVaultCache/blob/a3023eafc38bae91ef9b72341cca1c6f3bb2b0c3/Sander.KeyVaultCache/KeyVaultCache.cs#L26 A `NullReferenceException` is thrown. In this case, an `ArgumentNullException` should be thrown. The parameter name should be set to the `paramName` pareameter of the `ArgumentNullException` constructor.

Similarly, at https://github.com/SanderSade/Sander.KeyVaultCache/blob/a3023eafc38bae91ef9b72341cca1c6f3bb2b0c3/Sander.KeyVaultCache/KeyFetcher.cs#L70  A `NullReferenceException` is thrown although some other, perhaps a custom application exception, should be thrown.

**Comment by SanderSade** [comment]
Good catch for the first - I will change this in a day or few. Not sure if it is worth to do a new release, though.

On second, I have a deep dislike of custom exceptions, due to a huge project that used custom exceptions as a way to [control the business logic](http://wiki.c2.com/?DontUseExceptionsForFlowControl). Perhaps [KeyVaultErrorException](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.keyvault.models.keyvaulterrorexception?view=azure-dotnet)?

**Comment by tsahi** [comment]
`KeyVaultErrorException` derives from `RestException`, which describes a REST error, which is not the case here. Regardless of your personal past experience, I still believe a custom exception is in place here.
